File: lab3/src/cpp/src/queue.cpp
#include "queue.hpp"

void Queue::push(std::string_view value) {
    m_list.push_tail(value);
}

std::string Queue::pop() {
    if (m_list.getSize() == 0) {
        return {};
    }
    std::string value = m_list.get(0);
    m_list.del_head();
    return value;
}

std::string Queue::front() const {
    return m_list.get(0);
}

size_t Queue::size() const noexcept {
    return m_list.getSize();
}

bool Queue::empty() const noexcept {
    return m_list.getSize() == 0;
}

std::string Queue::print() const {
    if (empty()) {
        return "[] (front->back)";
    }
    // Используем твой метод print() из SinglyList
    return m_list.print() + " (front->back)";
}

File: lab3/src/go/internal/queue.go
package internal

type Queue struct {
	list *SinglyList
}

func NewQueue() *Queue {
	return &Queue{
		list: NewSinglyList(),
	}
}

func (q *Queue) Push(value string) {
	q.list.PushTail(value)
}

func (q *Queue) Pop() string {
	if q.list.GetSize() == 0 {
		return ""
	}
	val := q.list.head.Data
	q.list.DelHead()
	return val
}

func (q *Queue) Front() string {
	if q.list.GetSize() == 0 {
		return ""
	}
	return q.list.head.Data
}

func (q *Queue) Size() int {
	return q.list.GetSize()
}

func (q *Queue) Empty() bool {
	return q.list.GetSize() == 0
}

func (q *Queue) Print() string {
	if q.Empty() {
		return "[] (front->back)"
	}
	return q.list.Print() + " (front->back)"
}